### DISCUSSION TITLE
mc4plus comunicazione eth

### DISCUSSION THREAD

**Discussion by Simone-7-7** [robotology/community#511] [Q&A] [answered]
Buongiorno,
ho installato robotology-superbuild sia sotto Windows 10 che Debian 10 (buster) per controllare un motore ed un solenoide attraverso la mc4plus. Ho settato l'IP del laptop a 10.0.1.104 con Subnet 255.255.255.0 ed ho avviato FirmwareUpdater per provare la comunicazione con la scheda. Sotto Windows 10 il programma parte ma non trova niente mentre Debian restituisce il seguente errore:
qt.qpa.screen: QXcbConnection: Could not connect to display
Could not connect to any X display

Grazie
![FirmwareUpdater](https://user-images.githubusercontent.com/83953166/120630065-bb1ed380-c466-11eb-9062-a421a4a8f70e.png)

**Comment by pattacini** [comment]
Buongiorno @Simone-7-7 e grazie per aver trasferito il ticket qui!

@maggia80 ha messo a disposizione dei nuovi file di configurazione della board MC4Plus per la gestione del motore e del solenoide in https://github.com/robotology/robots-configuration/pull/262. @davidetome li ha visionati ma dobbiamo ancora testarli sul setup fisico.

Stay tuned 🎧

**Reply by pattacini** [reply]
@Simone-7-7, ci siamo accorti che nella PR segnalata in precedenza mancava proprio il file corretto `firmwareupdater.ini` che @davidetome ha appena aggiunto tramite https://github.com/robotology/robots-configuration/pull/268/files.

**Reply by Simone-7-7** [reply]
ho aggiunto il file firmwareupdater.ini ma ancora non riesco a vedere il device. C'è altro che dovrei fare? Allego screenshot del firmwareupdater.
![FirmwareUpdater01](https://user-images.githubusercontent.com/83953166/120646985-62593600-c47a-11eb-8761-6b8e5aed7b08.png)

**Comment by maggia80** [comment, marked as answer]
@Simone-7-7 dopo aver fatto Discover non trova alcuna board?

**Reply by Simone-7-7** [reply]
No, nessuna. Ho provato anche con Debian

**Reply by maggia80** [reply]
puoi allegare una foto della board? hai fatto la prova di collegare la board su entrambi i connettori ethernet?

**Reply by Simone-7-7** [reply]
Si ho provato
![20210603_154358](https://user-images.githubusercontent.com/83953166/120655558-16f75580-c483-11eb-99ab-d3ee78ad738d.jpg)

**Reply by maggia80** [reply]
sembra che il connettore della rete sia cablato al contrario. Il pin 5 dovrebbe essere il pin in cui c'è la calza mentre seguendo il cavo (se sotto la breadboard i cavi non sono incrociati) è nel pin 1
![image](https://user-images.githubusercontent.com/6987330/120657054-6ee28c00-c484-11eb-87e1-6b73f39f4d91.png)

qui noi usiamo nero e ross e bianco e verde come doppini.

**Reply by Simone-7-7** [reply]
Ho appena invertito i cavi, adesso abbiamo sul pin1 TXP pin2 TXN pin3 RXP pin4 RXN pin5 CT.
Con questa configurazione non mi vede neanche il cavo collegato
![Immagine](https://user-images.githubusercontent.com/83953166/120660817-009fc880-c488-11eb-89ad-38cda8425288.png)
![20210603_162047](https://user-images.githubusercontent.com/83953166/120660820-01385f00-c488-11eb-845e-21036b939d94.jpg)

**Reply by maggia80** [reply]
pin1 = white/green
pin2 = green
pin3 = white/orange
pin4 = orange
pin5 = yellow.

**Reply by maggia80** [reply]
@Simone-7-7

**Reply by Simone-7-7** [reply]
@maggia80 Stesso problema, non vede il cavo collegato e non trovo alcun dispositivo con il FirmwareUpdater

**Reply by maggia80** [reply]
anche collegando il connettore al P13?

**Reply by Simone-7-7** [reply]
@maggia80 Si succede su entrambi i connettori mentre se la calza si trova sul pin1 vede il cavo collegato ma comunque non trovo alcun disposito tramite firmwareupdater. Abbiamo ricontrollato le connessioni. Sotto la board i cavi non sono incrociati

**Reply by maggia80** [reply]
![image](https://user-images.githubusercontent.com/6987330/120780611-93903f80-c528-11eb-8e24-a82ecacca772.png)

scoperto il problema. C'era una goccia di stagno che metteva in corto diversi componenti. Risolto il problema la scheda pinga (10.0.1.99)

**Comment by Simone-7-7** [comment]
Buongiorno,
quando eseguo "yarpinterface --from yarpinterface.ini" il programma restituisce il seguente errore:
[ERROR] EthResource::verifyBoardTransceiver() detected different mn protocol minor versions: local = 18 , remote = 15 : cannot proceed any further.
[ERROR] ACTION REQUIRED: BOARD mc4plus-loco with IP 10.0.1.22 needs a FW update.
Dovrei aggiornare il firmware con FirmwareUpdater? A quale versione?

**Reply by davidetome** [reply]
Ciao,

 devi eseguire l'upgrade del [fw](https://github.com/robotology/icub-firmware-build/tree/master/ETH/MC4PLUS/bin/application) della `mc4plus`  utilizzando `FirmwareUpdater`.

- scarichi la repo
- esegui `FirmwareUpdater -a`
- Selezioni la board e cliccki su Force Maintenance mode
- e poi esegui Upload application
- alla fine riavvii la board

**Reply by davidetome** [reply]
Side note : basta eseguire `yarprobotinterface` da dentro la cartella del setup

**Reply by Simone-7-7** [reply]
Sembra che non riesca ad aggiornarsi. Di seguito gli screenshot del FirmwareUpdater interfaccia grafica e riga di comando
![Schermata a 2021-06-08 13-56-28](https://user-images.githubusercontent.com/83953166/121181164-fef84b00-c861-11eb-950a-d50f8f5955b7.png)
![Schermata a 2021-06-08 13-58-56](https://user-images.githubusercontent.com/83953166/121181171-00297800-c862-11eb-80d4-c99f43b884d9.png)
E' in questa situazione da più di 15 minuti

**Reply by davidetome** [reply]
dovrebbe metterci qualche secondo, hai selezionato il file `.hex` che ti ho puntato ovviamente, giusto?

**Reply by Simone-7-7** [reply]
Ho scaricato di nuovo il file e l'update è andato a buon fine, così come yarpinterface. Grazie

**Comment by Simone-7-7** [comment]
avviato yarpserver, yarprobotinterface e yarpmotorgui e riesco a vedere la posizione del motore ma non riesco a muoverlo. è stato fatto anche il corto al pin del fault. di seguito lo screenshot del yarpmotorgui
![Schermata a 2021-06-08 15-14-10](https://user-images.githubusercontent.com/83953166/121196862-1db20e00-c871-11eb-93f2-1e86f17514c7.png)
c'è qualcosa che ho omesso di fare?

**Reply by Simone-7-7** [reply]
Parlo degli encoder dei motori, non dei valori che si vedono nei joint dei solenoidi

**Reply by davidetome** [reply]
Ciao,

Ho creato un [fork](https://github.com/davidetome/robots-configuration) personale di `robots-configuration` dove ho fatto un paio di modifiche; ti chiedo di clonare ed utilizzare il mio fork visto che non saranno le uniche, poi mergeremo alla fine.

Prova per favore, ora il motore dovrebbe muoversi.

cc @ale-git @maggia80 @pattacini

**Reply by Simone-7-7** [reply]
ciao,
ancora non riesco a muovere il motore
![Schermata a 2021-06-09 16-08-01](https://user-images.githubusercontent.com/83953166/121370629-2a9c3380-c93d-11eb-8fcf-637db34d2f8c.png)
![Schermata a 2021-06-09 16-10-42](https://user-images.githubusercontent.com/83953166/121370757-443d7b00-c93d-11eb-82e7-2473d0964b9c.png)
dopo un po' di tempo che dice calibrating passa di nuovo a not configured
